src/docfx/build/toc/BuildTableOfContents.cs

```diff
@@ -11,26 +11,35 @@ namespace Microsoft.Docs.Build
 {
     internal static class BuildTableOfContents
     {
-        public static (IEnumerable<Error>, TableOfContentsModel, DependencyMap) Build(
-            Context context, Document file, TableOfContentsMap tocMap)
+        public static (IEnumerable<Error>, TableOfContentsModel, DependencyMap, List<string>) Build(
+            Context context, Document file, TableOfContentsMap tocMap, Dictionary<Document, List<string>> monikersMap)
         {
             Debug.Assert(file.ContentType == ContentType.TableOfContents);
 
             if (!tocMap.Contains(file))
             {
-                return (Enumerable.Empty<Error>(), null, DependencyMap.Empty);
+                return (Enumerable.Empty<Error>(), null, DependencyMap.Empty, new List<string>());
             }
 
             // todo: build resource files linked by toc
             var dependencyMapBuilder = new DependencyMapBuilder();
-            var (errors, tocModel, tocMetadata, refArticles, refTocs) = Load(context, file, dependencyMapBuilder);
+            var (errors, tocModel, tocMetadata, refArticles, refTocs, monikers) = Load(context, file, dependencyMapBuilder, monikersMap);
 
-            var model = new TableOfContentsModel { Items = tocModel, Metadata = file.Docset.Metadata.GetMetadata(file, tocMetadata) };
+            var metadata = file.Docset.Metadata.GetMetadata(file, tocMetadata);
 
-            return (errors, model, dependencyMapBuilder.Build());
+            // Monikers of toc file is the collection of every node's monikers
+            metadata["monikers"] = new JArray(monikers);
```

Comment:
Make `TableOfContentsMapModel.Metadata` of type `FileMetadata` to avoid `'monikers'`

Reply:
`filematadata` doesn't contains `monikers`, I will create a new model

---
In reply to: [235842238](https://github.com/dotnet/docfx/pull/3655#discussion_r235842238) [](ancestors = 235842238)